internal/storage/v2/clickhouse/tracestore/dbmodel/to.go

```diff
@@ -0,0 +1,142 @@
+	sr.DoubleAttributeKeys = append(sr.DoubleAttributeKeys, a.doubleKeys...)
+	sr.DoubleAttributeValues = append(sr.DoubleAttributeValues, a.doubleValues...)
+	sr.IntAttributeKeys = append(sr.IntAttributeKeys, a.intKeys...)
+	sr.IntAttributeValues = append(sr.IntAttributeValues, a.intValues...)
+	sr.StrAttributeKeys = append(sr.StrAttributeKeys, a.strKeys...)
+	sr.StrAttributeValues = append(sr.StrAttributeValues, a.strValues...)
+	sr.ComplexAttributeKeys = append(sr.ComplexAttributeKeys, a.complexKeys...)
+	sr.ComplexAttributeValues = append(sr.ComplexAttributeValues, a.complexValues...)
+}
+
+func (sr *SpanRow) appendEvent(event ptrace.SpanEvent) {
+	sr.EventNames = append(sr.EventNames, event.Name())
+	sr.EventTimestamps = append(sr.EventTimestamps, event.Timestamp().AsTime())
+
+	evAttrs := extractAttributes(event.Attributes())
+	sr.EventBoolAttributeKeys = append(sr.EventBoolAttributeKeys, evAttrs.boolKeys)
+	sr.EventBoolAttributeValues = append(sr.EventBoolAttributeValues, evAttrs.boolValues)
+	sr.EventDoubleAttributeKeys = append(sr.EventDoubleAttributeKeys, evAttrs.doubleKeys)
+	sr.EventDoubleAttributeValues = append(sr.EventDoubleAttributeValues, evAttrs.doubleValues)
+	sr.EventIntAttributeKeys = append(sr.EventIntAttributeKeys, evAttrs.intKeys)
+	sr.EventIntAttributeValues = append(sr.EventIntAttributeValues, evAttrs.intValues)
+	sr.EventStrAttributeKeys = append(sr.EventStrAttributeKeys, evAttrs.strKeys)
+	sr.EventStrAttributeValues = append(sr.EventStrAttributeValues, evAttrs.strValues)
+	sr.EventComplexAttributeKeys = append(sr.EventComplexAttributeKeys, evAttrs.complexKeys)
+	sr.EventComplexAttributeValues = append(sr.EventComplexAttributeValues, evAttrs.complexValues)
+}
+
+func (sr *SpanRow) appendLink(link ptrace.SpanLink) {
+	sr.LinkTraceIDs = append(sr.LinkTraceIDs, link.TraceID().String())
+	sr.LinkSpanIDs = append(sr.LinkSpanIDs, link.SpanID().String())
+	sr.LinkTraceStates = append(sr.LinkTraceStates, link.TraceState().AsRaw())
+
+	linkAttrs := extractAttributes(link.Attributes())
+	sr.LinkBoolAttributeKeys = append(sr.LinkBoolAttributeKeys, linkAttrs.boolKeys)
+	sr.LinkBoolAttributeValues = append(sr.LinkBoolAttributeValues, linkAttrs.boolValues)
+	sr.LinkDoubleAttributeKeys = append(sr.LinkDoubleAttributeKeys, linkAttrs.doubleKeys)
+	sr.LinkDoubleAttributeValues = append(sr.LinkDoubleAttributeValues, linkAttrs.doubleValues)
+	sr.LinkIntAttributeKeys = append(sr.LinkIntAttributeKeys, linkAttrs.intKeys)
+	sr.LinkIntAttributeValues = append(sr.LinkIntAttributeValues, linkAttrs.intValues)
+	sr.LinkStrAttributeKeys = append(sr.LinkStrAttributeKeys, linkAttrs.strKeys)
+	sr.LinkStrAttributeValues = append(sr.LinkStrAttributeValues, linkAttrs.strValues)
+	sr.LinkComplexAttributeKeys = append(sr.LinkComplexAttributeKeys, linkAttrs.complexKeys)
+	sr.LinkComplexAttributeValues = append(sr.LinkComplexAttributeValues, linkAttrs.complexValues)
+}
+
+func extractAttributes(attrs pcommon.Map) (out struct {
```

Comment:
I recommend defining a struct for the output value and using it inside SpanRow:
```go
type Attributes struct {
	BoolKeys      []string
	BoolValues    []bool
	...
}

type SpanRow struct {
	...
	Attributes attributes // instead of many fields
}
```
and similarly for links/events
```
type Attributes2D struct {
	BoolKeys      [][]string
	BoolValues    [][]bool
	...
}
type SpanRow struct {
	...
	Attributes attributes // instead of many fields
	Attributes2D eventAttributes
	Attributes2D linkAttributes
}
```